### ISSUE TITLE
All latest tools fail to authenticate to GCS

### ISSUE THREAD

**Issue by shnapz** [spotify/gcs-tools#26]
STR:

1a. Install all latest (v0.2.2 on Aug 29) tools
1b. Or build latest master to parquet-cli-1.12.3.jar, proto-tools-3.21.1.jar, avro-tools-1.11.0.jar,magnolify-tools-0.4.8.jar

2. Run all of them using basic read command like `<TOOL> tojson <GCS_PATH>`

**Actual:**
Tool launches browser that shows a page:
<img width="476" alt="Screen Shot 2022-08-29 at 9 50 20 AM" src="https://user-images.githubusercontent.com/942930/187216827-d0b811e8-2b0e-4eb8-8f7f-f47d19b77837.png">

With a message:

> The version of the app you're using doesn't include the latest security features to keep you protected. Please make sure to download from a trusted source and update to the latest, most secure version.

**Exected:**
Tool reads a file according to spec

**Comment by shnapz** [comment]
The auth is failing because Google deprecated support of **OAuth out-of-band (oob) flow**:
https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#dates-oob

Although Oct 3, 2022 is a date when existing clients will be deprecated, currently auth is not working for gcs-tools

Solution: move to a new OAuth flow, by upgrading to "com.google.cloud.bigdataoss" % "gcs-connector" % "hadoop3-2.2.7" and changing hadoop connector configuration

**Comment by shnapz** [comment]
Blocked by https://github.com/GoogleCloudDataproc/hadoop-connectors/issues/864

**Comment by jwiklund** [comment]
A workaround using application default credentials is https://github.com/spotify/gcs-tools/compare/main...jwiklund:gcs-tools:update?expand=1 (I don't know enough about other use cases to tell if this makes sense, but it works "on my machine").

**Comment by shnapz** [comment]
Thanks @jwiklund 
I assume you are using service account when accessing GCS?
```
    <name>fs.gs.auth.service.account.enable</name>
    <value>true</value>
    <description>Force OAuth2 flow</description>
```
You set:
```
export GOOGLE_APPLICATION_CREDENTIALS=$HOME/.config/gcloud/application_default_credentials.json
```
Did you specifically modify your default credentials?

**Comment by jwiklund** [comment]
No, I've only done `gcloud auth application-default login`, but there is some code in the connector that doesn't check for the default location unless that environment variable is present (it times out checking for the metadata service instead). If possible it might be useful to check if that file exist and only enable service.account.enable then (or I guess implement our own provider that uses the standard google auth lookup library).

**Comment by jwiklund** [comment]
Actually I don't have to use that env var on my other machine, something is different here so that it picks it up directly.

**Comment by jwiklund** [comment]
Ok, now it happened again, so "sometimes" you have to include the GOOGLE_APPLICATION_CREDENTIALS environment variable or else it will fallback to metadata lookup (which fails on non gce/gke machines). I'm not sure what triggers the different behavior.

**Comment by shnapz** [comment]
User credentials auth will be supported [in 3.0.0 release in 2023 Q1](https://github.com/GoogleCloudDataproc/hadoop-connectors/issues/864#issuecomment-1367532319)

**Comment by shnapz** [comment]
FYI they just closed the ticket and committed to a fix in 3.0, but 3.0 is not there yet, we don't have estimations

**Comment by RustedBones** [comment]
Fixed in 0.3.0
